src/applications/vaos/new-appointment/components/DateTimeSelectPage/index.jsx

```diff
@@ -217,69 +302,54 @@ export default function DateTimeSelectPage() {
         </span>
       </h1>
       {!loadingSlots && (
-        <WaitTimeAlert
-          eligibleForRequests={eligibleForRequests}
-          facilityId={facilityId}
-          nextAvailableDate={new Date(availableSlots?.[0]?.start)}
-          preferredDate={parseISO(preferredDate)}
-          timezone={timezone}
-        />
-      )}
-      {fetchFailed && (
-        <ErrorMessage
+        <AlertSection
+          fetchStatus={appointmentSlotsStatus}
           history={history}
-          facilityId={facilityId}
-          requestAppointmentDateChoice={(...args) =>
-            dispatch(requestAppointmentDateChoice(...args))
-          }
-          eligibility={eligibility}
+          requestEligible={eligibleForRequests}
+          slotAvailable={slotAvailable}
+          earliestDate={earliestDate}
+          preferredDate={startMonth}
+          timezone={timezone}
         />
       )}
-      {!fetchFailed && (
-        <>
-          <p>
-            {clinic && `Scheduling at ${clinic.serviceName}`}
-            {clinic && timezone && <br />}
-            {timezone && `Times are displayed in ${timezoneDescription}.`}
-          </p>
-          <CalendarWidget
-            maxSelections={1}
-            availableSlots={availableSlots}
-            value={selectedDates}
-            id="dateTime"
-            timezone={timezone}
-            additionalOptions={{
-              required: true,
-            }}
-            disabled={loadingSlots}
-            hideWhileDisabled
-            disabledMessage={
-              // eslint-disable-next-line react/jsx-wrap-multilines
-              <va-loading-indicator
-                data-testid="loadingIndicator"
-                set-focus
-                message="Finding appointment availability..."
-                label="Finding appointment availability"
-              />
-            }
-            onChange={(...args) => dispatch(onCalendarChange(...args))}
-            onNextMonth={(...args) => dispatch(getAppointmentSlots(...args))}
-            onPreviousMonth={(...args) =>
-              dispatch(getAppointmentSlots(...args))
-            }
-            minDate={addDays(new Date(), 1)}
-            maxDate={addDays(new Date(), 395)}
-            renderIndicator={_ => undefined}
-            required
-            requiredMessage="Please choose your preferred date and time for your appointment"
-            startMonth={startMonth}
-            showValidation={submitted && !selectedDates?.length}
-            showWeekends
-            upcomingAppointments={upcomingAppointments}
-            isAppointmentSelectionError={isAppointmentSelectionError}
-          />
-        </>
-      )}
+      {!fetchFailed &&
+        (loadingSlots || slotAvailable) && (
```

Comment:
Is my understanding correct here? If the fetch is ok and we are loadingSlots or we have some slots, display the calendar? I don't understand how loadingSlots is used. I see loadingSlots is set here:

```
const isInitialLoad = useIsInitialLoad(loadingSlots);
```

Reply:
Yes. Bullet 3 was this missing piece for me. Thanks!